### PR TITLE
 [DOCS] Adds data frame analytics overview

### PR DESCRIPTION
**Pull request by szabosteve**
This PR adds the data frame analysis overview under the ML section.

Related issue: https://github.com/elastic/ml-team/issues/175

### CODE REVIEW

File: docs/en/stack/ml/dataframeanalytics.asciidoc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+and clean your data by using the 
+{ref}/data-frame-apis.html[{dataframe-transform}] feature.
+
+[float]
+[[dfa-outlier-detection]]
+==== {oldetection-cap}
+
+{oldetection-cap} is an analysis for identifying data points with unusual 
+features (outliers) that deviates from the features of the majority of the data 
+points in a particular dataset. Outliers may denote errors or unusual behavior.
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/en/stack/ml/dataframeanalytics.asciidoc]: Agree. BTW, I referred to the points as outliers. Later I added the part about the features and forgot to replace the bracketed text, so thank you for spotting this.

File: docs/en/stack/ml/dataframeanalytics.asciidoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
+[role="xpack"]
+[[ml-dfanalytics]]
+=== {dfanalytics-cap}
+
+{dfanalytics-cap} allows you to analyze your data with different types of
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/en/stack/ml/dataframeanalytics.asciidoc]: I take it. It's better.

File: docs/en/stack/ml/dataframeanalytics.asciidoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+We use unsupervised {oldetection} which means that it doesn't use training 
+dataset for teaching the {oldetection} model to recognize outliers. Unsupervised 
+{oldetection} uses various machine learning techniques to find which data points 
+are unusual compared to the majority of the data points.
+
+In the {stack}, we use four {oldetection} methods. The basic assumption of the 
+**distance based methods** is that normal data points – in other words, points 
+that are not outliers – have a lot of neighbors nearby, because we expect that 
+in a population the majority of the data points have similar features, while the
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/en/stack/ml/dataframeanalytics.asciidoc]: That's the distinction that I needed! Thanks!

File: docs/en/stack/ml/dataframeanalytics.asciidoc [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+{dfanalytics-cap} allows you to perform different analyses of your data and 
+annotate it with the results. Essentially, as part of its output, {dfanalytics} 
+appends the results of the analysis to the source data. By doing this, it 
+provides additional insights on the data. The process leaves the source index 
+intact, it creates a new index that contains the annotated data. You can slice 
+and dice the data extended with the results as you normally do with any other 
+data set.
+
+IMPORTANT: Using {dfanalytics} assumes you have already done some preprocessing
```
Comment (sophiec20) [inline docs/en/stack/ml/dataframeanalytics.asciidoc]: We do not assume that data frame transforms have been used. Please change to something along the lines of ...

`Using {dfanalytics} requires source data to be structured as a two dimensional "tabular" data structure i.e. a data frame. Data frame transforms allow you to easily create data frames which can be used as the source for data frame analytics.`

File: docs/en/stack/ml/dataframeanalytics.asciidoc [hunk 5]
```diff
@@ -0,0 +1,98 @@
+[role="xpack"]
+[[ml-dfanalytics]]
+=== {dfanalytics-cap}
+
+{dfanalytics-cap} allows you to perform different analyses of your data and 
+annotate it with the results. Essentially, as part of its output, {dfanalytics} 
+appends the results of the analysis to the source data. By doing this, it 
+provides additional insights on the data. The process leaves the source index 
+intact, it creates a new index that contains the annotated data. You can slice
```
Comment (sophiec20) [inline docs/en/stack/ml/dataframeanalytics.asciidoc]: `... it creates a new index that contains a copy of the source data and the annotated data.`

**Comment by lcawl** [comment]
I've rebased this PR to pick up the structural changes I implemented in the master branch. You can preview this content here: http://stack-docs_383.docs-preview.app.elstc.co/guide/en/elastic-stack-overview/master/ml-dfanalytics-overview.html

File: docs/en/stack/ml/df-analytics/dataframeanalytics.asciidoc [hunk 6]
```diff
@@ -0,0 +1,103 @@
+[role="xpack"]
+[[ml-dfanalytics-overview]]
+== {dfanalytics-cap} overview
+++++
+<titleabbrev>Overview</titleabbrev>
+++++
+
+{dfanalytics-cap} allows you to perform different analyses of your data and
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dataframeanalytics.asciidoc]: I stole this introductory sentence for the following page: https://www.elastic.co/guide/en/elastic-stack-overview/master/ml-dfanalytics.html
Should all of the content in this introductory section be moved to that page (akin to what was done in the Data frame transforms page)? If so, "Outlier detection" would be the first sub-page, which seems okay to me.

**Pull request merged**